sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -67,23 +66,25 @@
     /**
      * Creates a CryptographyAsyncClient that uses {@code pipeline} to service requests
      *
-     * @param key the JsonWebKey to use for cryptography operations.
+     * @param key the key to use for cryptography operations.
      * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
      * @param version {@link CryptographyServiceVersion} of the service to be used when making requests.
      */
-    CryptographyAsyncClient(JsonWebKey key, HttpPipeline pipeline, CryptographyServiceVersion version) {
-        Objects.requireNonNull(key);
-        if (!key.isValid()) {
+    CryptographyAsyncClient(KeyVaultKey key, HttpPipeline pipeline, CryptographyServiceVersion version) {
+        Objects.requireNonNull(key, "The key vault key is required.");
+        JsonWebKey jsonWebKey = key.getKey();
+        Objects.requireNonNull(jsonWebKey, "The Json web key is required.");
+        if (!jsonWebKey.isValid()) {
             throw new IllegalArgumentException("Json Web Key is not valid");
         }
-        if (key.getKeyOps() == null) {
+        if (jsonWebKey.getKeyOps() == null) {
             throw new IllegalArgumentException("Json Web Key's key operations property is not configured");
         }
 
         if (key.getKeyType() == null) {
             throw new IllegalArgumentException("Json Web Key's key type property is not configured");
         }
-        this.key = key;
+        this.key = jsonWebKey;
```

Comment:
You probably want to store the entire KeyVaultKey so that you can validate the current time is in between notBefore expiresOn when performing encrypt, sign, and wapKey. But you could file an issue on this now and do a separate PR for it.